stuffer/s2n_stuffer.h

```diff
@@ -165,6 +166,16 @@ int s2n_stuffer_skip_read_until(struct s2n_stuffer *stuffer, const char *target)
 int s2n_stuffer_alloc_ro_from_string(struct s2n_stuffer *stuffer, const char *str);
 int s2n_stuffer_init_ro_from_string(struct s2n_stuffer *stuffer, uint8_t *data, uint32_t length);
 
+/* Stuffer versions of sprintf methods, except:
+ * - They write bytes, not strings. They do not write a final '\0'. Unfortunately,
+ *   they do still require enough space for a final '\0'-- we'd have to reimplement
+ *   sprintf to avoid that.
+ * - vprintf does not consume the vargs. It calls va_copy before using
+ *   the varg argument, so can be called repeatedly with the same vargs.
+ */
+int s2n_stuffer_printf(struct s2n_stuffer *stuffer, const char *format, ...);
+int s2n_stuffer_vprintf(struct s2n_stuffer *stuffer, const char *format, va_list vargs);
```

Comment:
IIRC all of the stuffer functions have CBMC proofs. It's probably good to have some for these new functions as well.

Reply:
Definitely agree, but I think getting CBMC to work with varargs is going to be a bit of an adventure. I'd rather that didn't block this PR, particularly because the only direct manipulation of the stuffer is the "tainted" flag logic. Other than that, we just use existing stuffer functions.